algorithms/curiosity/curiosity_model.py

```diff
@@ -0,0 +1,87 @@
+import tensorflow as tf
+
+from algorithms.encoders import make_encoder
+from algorithms.env_wrappers import has_image_observations, main_observation_space
+from algorithms.tf_utils import dense, count_total_parameters, conv
+from utils.utils import log
+
+
+class CuriosityModel:
+    """Single class for inverse and forward dynamics model."""
+
+    def __init__(self, env, obs, next_obs, actions, past_frames, forward_fc, params=None):
+        """
+        :param obs - placeholder for observations
+        :param actions - placeholder for selected actions
+        """
+        with tf.variable_scope('curiosity_model'):
+            self.regularizer = tf.contrib.layers.l2_regularizer(scale=1e-10)
+
+            obs_space = main_observation_space(env)
+            image_obs = has_image_observations(obs_space)
+            num_actions = env.action_space.n
+
+            if image_obs:
```

Comment:
I think function make_encoder can handle both image and low-dimensional stuff. Can you remove the low-dimensional branch?